### PR TITLE
DDF-2565 upgrade camel 2.18  and artemis 1.4

### PR DESCRIPTION
**Pull request by ryeats**
#### What does this PR do?
upgrade artemis to 1.4.0 and upgrades camel to 2.18.0 for some bugfixes and to get the latest camel-spring so we can use the caching connection factory with artemis
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@roelens8 @jrnorth @harrison-tarr 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Core APIs](https://github.com/orgs/codice/teams/core-apis)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@pklinef
#### How should this be tested? (List steps with links to updated documentation)
run integratikon tests make sure version changes didnt break anything
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by roelens8** [comment]
👍

File: platform/platform-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -1079,11 +1123,16 @@
...
+        <bundle>mvn:org.apache.geronimo.specs/geronimo-json_1.0_spec/${geronimo-json-spec.version}
+        </bundle>
+        <bundle>mvn:org.apache.johnzon/johnzon-core/${johnzon.version}</bundle>
+
         <bundle>mvn:org.apache.activemq/artemis-native/${artemis.version}</bundle>
         <bundle>mvn:org.apache.activemq/artemis-server-osgi/${artemis.version}</bundle>
     </feature>
 
-    <feature name="artemis-amqp" install="manual" version="${artemis.version}" description="ActiveMQ Artemis AMQP protocol libraries">
+    <feature name="artemis-amqp" version="${pom.version}"
```
Comment (jrnorth) [inline platform/platform-app/src/main/resources/features.xml]: Why is this `${pom.version}` now?

File: pom.xml [hunk 2]
```diff
@@ -91,33 +90,44 @@
...
-        <geronimo-jta_1.1_spec.version>1.1.1</geronimo-jta_1.1_spec.version>
-        <geronimo-jms_1.1_spec.version>1.1.1</geronimo-jms_1.1_spec.version>
+        <commons.fileupload.version>1.3.2</commons.fileupload.version>
+        <logback.classic.version>1.1.7</logback.classic.version>
+        <!--camel dependencies-->
+        <camel.version>2.18.0</camel.version>
+        <servicemix-specs.version>2.7.0</servicemix-specs.version>
+        <geronimo-jta-spec.version>1.1.1</geronimo-jta-spec.version>
+        <geronimo-jms-spec.version>1.1.1</geronimo-jms-spec.version>
+        <commons-lang.version>2.6</commons-lang.version>
```
Comment (jrnorth) [inline pom.xml]: Looks like a few of these are duplicates. `commons-lang` and `commons-collections` are defined above.

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (roelens8) [inline platform/platform-app/src/main/resources/features.xml]: This feature should still have the install attribute equal to "manual".

File: platform/platform-app/src/main/resources/features.xml [hunk 3]
```diff
@@ -1092,6 +1141,12 @@
         <bundle>mvn:org.apache.activemq/artemis-amqp-protocol/${artemis.version}</bundle>
     </feature>
 
+    <feature name="artemis-mqtt" version="${pom.version}"
```
Comment (roelens8) [inline platform/platform-app/src/main/resources/features.xml]: This feature should still have the install attribute equal to "manual".

Comment (roelens8) [inline platform/platform-app/src/main/resources/features.xml]: This feature also has the version set to ${pom.version}

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (ryeats) [inline platform/platform-app/src/main/resources/features.xml]: copy and paste error thought i fixed all of them

File: pom.xml (hunk 2 above)
Comment (ryeats) [inline pom.xml]: removed the duplicates

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (ryeats) [inline platform/platform-app/src/main/resources/features.xml]: fixed

**Pull request merged**